packages/embark-coverage/src/lib/injector.ts

```diff
@@ -0,0 +1,85 @@
+import { ContractEnhanced } from "./contractEnhanced";
+import { encrypt } from "./eventId";
+import { InjectionPoint } from "./types";
+
+import * as semver from "semver";
+import parser, { ASTNode, ContractDefinition, EmitStatement, EventDefinition, ExpressionStatement } from "solidity-parser-antlr";
+
+const EMIT_VERSION =  "0.4.21";
+
+const EVENT_NODE: EventDefinition = {
+  isAnonymous: false,
+  name: "__StatementCoverage",
+  parameters: {
+    parameters: [
+      {
+        isIndexed: false,
+        isStateVar: false,
+        name: "value",
+        type: "VariableDeclaration",
+        typeName: {
+          name: "uint32",
+          type: "ElementaryTypeName",
+        },
+      },
+    ],
+    type: "ParameterList",
+  },
+  type: "EventDefinition",
+};
+
+export class Injector {
+  private isEmitSupported: boolean;
+
+  constructor(private contract: ContractEnhanced, private solcVersion: string) {
+    this.solcVersion = solcVersion;
+    this.isEmitSupported = semver.gte(this.solcVersion, EMIT_VERSION);
+  }
+
+  public process(injectionPoint: InjectionPoint) {
+    switch (injectionPoint.type) {
+      case "statement":
+        return this.statement(injectionPoint);
+      case "contractDefinition":
+        return this.contractDefinition(injectionPoint);
+    }
+  }
+
+  private emitStatementNode(id: number): ASTNode {
+    // Depending on what solidity version we're handling, "emit" statements might not
+    // be supported, so we check here.
+    const functionCall = {
+      arguments: [{ number: id.toString(), type: "NumberLiteral" }],
+      expression: { name: "__StatementCoverage", type: "Identifier" },
+      names: [],
+      type: "FunctionCall",
+    };
```

Comment:
It probably make sense to move `functionCall` out of this function, because we don't need to reinitialize every time `emitStatementNode()` is called.

Reply:
Good catch 👏

Reply:
I just realized why I did it this way: the arguments to the function call change, so we have to initialize it inside.

Reply:
Makes sense!